### ISSUE TITLE
Incorrect prices returned in joined GQL operation

### ISSUE THREAD

**Issue by magento-engcom-team** [magento/magento2#30176]
In PWA Studio we have a [mutation composed of two mutations](https://github.com/magento/pwa-studio/blob/develop/packages/venia-ui/lib/components/CartPage/ProductListing/EditModal/productForm.js#L137-L171). I found that when requesting a "additional*payment_methods" in both response objects the grand*total in the second mutation response body comes back as the total from after the first mutation.

Here's a contrived example of the mutations.

```
mutation updateItem {
  addItem(...) {
    cart {
      additional*payment*methods
    }
  }
  removeItem(...) {
    cart {
      additional*payment*methods
      prices {
        grand_total
      }
    }
  }
}
```

See this comment for more detail: https://github.com/magento/pwa-studio/pull/2394#issuecomment-632908862

**Comment by magento-engcom-team** [comment]
The issue was exported from the internal JIRA. The link to the original JIRA issue: https://jira.corp.magento.com/browse/MC-34586

**Comment by mauragcyrus** [comment]
Not straight forward or a high priority issue, the PWA team currently has a work around.

**Comment by shikhamis11** [comment]
@magento give me 2.4-develop instance

**Comment by shikhamis11** [comment]
hey 
I am checking below query 
```
mutation UpdateConfigurableOptions(
  $cartId: String!
  $cartItemId: Int!
  $parentSku: String!
  $variantSku: String!
  $quantity: Float!
) {
  addConfigurableProductsToCart(
    input: {
      cart_id: $cartId
      cart_items: [
        {
          data: { quantity: $quantity, sku: $variantSku }
          parent_sku: $parentSku
        }
      ]
    }
  ) {
    cart {
      id
      total_quantity
      available_payment_methods {
        code
        title
      }
      prices {
        grand_total {
          value
          currency
        }
      }
    }
  }
  removeItemFromCart(input: { cart_id: $cartId, cart_item_id: $cartItemId }) {
    cart {
      id
      total_quantity
      available_payment_methods {
        code
        title
      }
      prices {
        grand_total {
          value
          currency
        }
      }
    }
  }
}
`
```
variables

```
{
  "cartId":"HrDsuEYzcAfHDtP7Cy09Vhf0gbI0VgVU",
  "cartItemId":13,
  "parentSku":"WJ01",
  "variantSku":"WJ01-S-Red",
  "quantity":1
}
```


I am getting response 
```
{
  "data": {
    "addConfigurableProductsToCart": {
      "cart": {
        "id": "HrDsuEYzcAfHDtP7Cy09Vhf0gbI0VgVU",
        "total_quantity": 2,
        "available_payment_methods": [
          {
            "code": "checkmo",
            "title": "Check / Money order"
          }
        ],
        "prices": {
          "grand_total": {
            "value": 150,
            "currency": "USD"
          }
        }
      }
    },
    "removeItemFromCart": {
      "cart": {
        "id": "HrDsuEYzcAfHDtP7Cy09Vhf0gbI0VgVU",
        "total_quantity": 1,
        "available_payment_methods": [
          {
            "code": "checkmo",
            "title": "Check / Money order"
          }
        ],
        "prices": {
          "grand_total": {
            "value": 75,
            "currency": "USD"
          }
        }
      }
    }
  }
}
```

**Comment by shikhamis11** [comment]
It seems it is working as expected

**Comment by shikhamis11** [comment]
I have experienced another error while checking this please check #31779

**Comment by mauragcyrus** [comment]
Not reproducible

**Comment by magento-engcom-team** [comment]
Hi @magento-engcom-team, @shikhamis11. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-34586` was closed as `Fixed`. 

The fix will be available with the upcoming `2.3.5` release.
